project/src/test/kotlin/org/ionproject/core/calendar/CalendarControllerTest.kt

```diff
@@ -0,0 +1,112 @@
+package org.ionproject.core.calendar
+
+
+import org.ionproject.core.common.Media
+import org.ionproject.core.common.Uri
+import org.ionproject.core.utils.ControllerTester
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+
+internal class CalendarControllerTest : ControllerTester() {
+    companion object {
+        val courseID = 1
+        val calTerm = "1718v"
+        val classSection = "1D"
+
+        val calendarByClass = """
+                BEGIN:VCALENDAR
+                PRODID:/v0/courses/1/classes/1718v
+                VERSION:2.0
+                BEGIN:VJOURNAL
+                UID:1
+                DTSTAMP:20200511T162630Z
+                SUMMARY;LANGUAGE=pt-PT:uma sinopse
+                SUMMARY;LANGUAGE=en-US:some summary
+                DESCRIPTION;LANGUAGE=en-US:this is a description
+                ATTACH:https://www.google.com
+                DTSTART:20200410T140000Z
+                CREATED:20200511T162630Z
+                CATEGORIES;LANGUAGE=pt-PT:EXAME
+                CATEGORIES;LANGUAGE=en-US:EXAM
+                CATEGORIES;LANGUAGE=en-GB:EXAM
+                END:VJOURNAL
+                BEGIN:VTODO
+                UID:2
+                DTSTAMP:20200511T162630Z
+                SUMMARY;LANGUAGE=en-US:do the thing
+                DESCRIPTION;LANGUAGE=en-US:I have to do the thing soon enough
+                ATTACH:https://www.google.com
+                CREATED:20200511T162630Z
+                DUE:20210619T140000Z
+                CATEGORIES;LANGUAGE=pt-PT:EXAME
+                CATEGORIES;LANGUAGE=en-US:EXAM
+                END:VTODO
+                BEGIN:VTODO
+                UID:3
+                DTSTAMP:20200511T162630Z
+                SUMMARY;LANGUAGE=en-US:don't do the thing
+                DESCRIPTION;LANGUAGE=en-US:I have to do the thing soon enough
+                CREATED:20200511T162630Z
+                DUE:20210619T140000Z
+                CATEGORIES;LANGUAGE=pt-PT:EXAME
+                CATEGORIES;LANGUAGE=en-US:EXAM
+                END:VTODO
+                BEGIN:VEVENT
+                UID:4
+                DTSTAMP:20200511T162630Z
+                SUMMARY;LANGUAGE=en-US:1st exam WAD
+                DESCRIPTION;LANGUAGE=en-US:Normal season exam for WAD-1920v
+                CREATED:20200511T162630Z
+                CATEGORIES;LANGUAGE=en-GB:EXAM
+                DTSTART:20200619T140000Z
+                DTEND:20200619T150000Z
+                END:VEVENT
+                END:VCALENDAR
+                """.replace("\\s".toRegex(), "")
```

Comment:
I'm not sure if it is a good idea to manipulate the strings before testing for equality. I suggest comparing the retrieved payload with the expected string as is, with no manipulation (trimming, replacement, etc)

Reply:
I had a hard time trying to make the strings match because of the newlines and spaces even if the contents were the same, my quick solution was to simply remove all spaces and newlines.

Reply:
I see. That probably has to do with [iCalendar's way of defining content lines](https://tools.ietf.org/html/rfc5545#section-3.1).